sdks/python/apache_beam/io/parquetio.py

```diff
@@ -0,0 +1,349 @@
+
+    Returns:
+      A WriteToParquet transform usable for writing.
+    """
+    super(WriteToParquet, self).__init__()
+    self._sink = \
+      _create_parquet_sink(
+          file_path_prefix,
+          schema,
+          codec,
+          row_group_size,
+          file_name_suffix,
+          num_shards,
+          shard_name_template,
+          mime_type
+      )
+
+  def expand(self, pcoll):
+    return pcoll | Write(self._sink)
+
+  def display_data(self):
+    return {'sink_dd': self._sink}
+
+
+def _create_parquet_sink(file_path_prefix,
+                         schema,
+                         codec,
+                         row_group_size,
+                         file_name_suffix,
+                         num_shards,
+                         shard_name_template,
+                         mime_type):
+  return \
+    _ParquetSink(
+        file_path_prefix,
+        schema,
+        codec,
+        row_group_size,
+        file_name_suffix,
+        num_shards,
+        shard_name_template,
+        mime_type
+    )
+
+
+class _ParquetSink(filebasedsink.FileBasedSink):
+  """A sink for parquet files."""
+
+  def __init__(self,
+               file_path_prefix,
+               schema,
+               codec,
+               row_group_size,
+               file_name_suffix,
+               num_shards,
+               shard_name_template,
+               mime_type):
+    super(_ParquetSink, self).__init__(
+        file_path_prefix,
+        file_name_suffix=file_name_suffix,
+        num_shards=num_shards,
+        shard_name_template=shard_name_template,
+        coder=None,
+        mime_type=mime_type,
+        # Compression happens at the block level using the supplied codec, and
+        # not at the file level.
+        compression_type=CompressionTypes.UNCOMPRESSED)
+    self._schema = schema
+    self._codec = codec
+    self._row_group_size = row_group_size
+    self._buffer = [[] for _ in range(len(schema.names))]
+    self._file_handle = None
+
+  def open(self, temp_path):
+    self._file_handle = super(_ParquetSink, self).open(temp_path)
+    return ParquetWriter(
+        self._file_handle, self._schema, compression=self._codec)
+
+  def write_record(self, writer, value):
+    if len(self._buffer[0]) >= self._row_group_size:
+      self._write_buffer(writer)
+
+    # reorder the data in columnar format.
+    for i, n in enumerate(self._schema.names):
+      self._buffer[i].append(value[n])
+
+  def close(self, writer):
+    if len(self._buffer[0]) > 0:
+      self._write_buffer(writer)
+
+    writer.close()
+    if self._file_handle:
+      self._file_handle.close()
+      self._file_handle = None
+
+  def display_data(self):
+    res = super(_ParquetSink, self).display_data()
+    res['codec'] = str(self._codec)
+    res['schema'] = str(self._schema)
+    return res
+
+  def _write_buffer(self, writer):
+    for x, y in enumerate(self._buffer):
+      self._buffer[x] = pa.array(y)
```

Comment:
Seems like you modify the buffer here to store an intermediate result. Please add a comment.

Probably cleaner to create a new list here instead of modifying the existing buffer to store something different.